### ISSUE TITLE
contributing to moin2.0: discussion on code complexity

### ISSUE THREAD

**Issue by sandeep-gh** [moinwiki/moin#1273]
This is mostly for discussion. I spend last two days looking at the codebase trying to understand its design and execution flow. This is just a feedback (feel free to ignore or use at will). While I am very happy that full-fledged wiki system exists in Python that can be tinkered, I find that codebase hard to follow and make changes.  I happen to prior come across a java codebase that I had similar difficulty. Moin codebase feels Javaish (for lack of better word) that Pythonish. By Javaish I mean classes and hierarchies that make it hard to follow the code flow. By Pythonish, I would say explicit and direct. 

To give a concrete example, I had very tough time making the mental map for Storage, Backend, Stores (fs, sqlite, sqla), and all the various abstract classes. 
I had full intentions of contributing but I am less enthusiastic now.  Don't have specific fixable issues to mention. Just wanted to communicate, in case if it helps the project. 

I did learned from Moin codebase and it is certainly very useful.

**Comment by RogerHaase** [comment]
I also find the code difficult to understand and wish there were more comments.

But that is the current state.  Contributions to make it better are welcome.

If there are specific defects that are fixable, please describe in separate issues. But creating an issue that says the whole codebase sucks, and wishing it were somehow better, cannot be fixed with one patch.
